src/generic/c1_curved_elements.h

```diff
@@ -1061,7 +1058,38 @@ namespace oomph
                         DShape& d2bpsi,
                         const DenseMatrix<double>& m) const; // PRIVATE
 
-      // HERE WRITE A PUBLIC d2shape_local
+     
+     /// Two by two specialisation of function to calculate inverse of a matrix
+     static inline double invert_two_by_two(const DenseMatrix<double>& jacobian,
+                                     DenseMatrix<double>& inverse_jacobian)
+      {
+       // Calculate the determinant of the matrix
+       const double det =
+        jacobian(0, 0) * jacobian(1, 1) - jacobian(0, 1) * jacobian(1, 0);
+       
+       // Report if Matrix is singular or negative
+#ifdef PARANOID
+       if (fabs(det) < 1e-12)
+        {
+         std::stringstream error_stream;
+         error_stream
+          << "The matrix seems to be singular : det(M) = " << det
+          << ".\n";
+         throw OomphLibError(
+          error_stream.str(), OOMPH_CURRENT_FUNCTION, OOMPH_EXCEPTION_LOCATION);
+        }
+#endif
+       
+       // Calculate the inverse of the 2x2 matrix
+      inverse_jacobian(0, 0) = jacobian(1, 1) / det;
+      inverse_jacobian(0, 1) = -jacobian(0, 1) / det;
+      inverse_jacobian(1, 0) = -jacobian(1, 0) / det;
+      inverse_jacobian(1, 1) = jacobian(0, 0) / det;
+      
+      return det;
+      }
+
+      // HERE WRITE A PUBLIC d2shape_local // hierher Aidan ?
```

Comment:
I think I did this years ago, and the commit has been abandoned... Not usually necessary for these elements as everything is in terms of global, this is meant to be for futureproofing.

Reply:
Delete comment

Reply:
Keeping this unresolved to alert Aidan to the fact that he also has to search "HERE" (in addition to "debug" and "[zdec]" when finding things to tidy. There's plenty of stuff in this file (.h and .cc). Looks to me as if the comments refer to possible but not strictly necessary rewrites, in which case I suggest we just get rid of the annotation and deem things to be OK as they are -- unless you spot easy gains by rewriting things (quickly).